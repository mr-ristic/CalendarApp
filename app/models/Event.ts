import { Instance, SnapshotIn, SnapshotOut, types } from 'mobx-state-tree';
import { withSetPropAction } from './helpers/withSetPropAction';
import { UserModel } from './User';

/**
 * EventModel represents an event with a specific start and end time.
 * @property {number} id - The unique identifier for an event.
 * @property {string} title - The title of the event.
 * @property {string} date - The date of the event (ISO formatted string).
 * @property {string} startTime - The start time of the event (24-hour format).
 * @property {string} endTime - The end time of the event (24-hour format).
 * @property {string} location - The location where the event will take place.
 * @property {boolean} isAllDay - Indicates whether the event lasts all day.
 * @property {types.reference} userId - A reference to the user associated with the event.
 */
export const EventModel = types
  .model('Event')
  .props({
    id: types.identifierNumber,
    title: types.string,
    date: types.string,
    startTime: types.string,
    endTime: types.string,
    location: types.string,
    isAllDay: types.boolean,
    userId: types.reference(UserModel)
  })
  .actions(withSetPropAction)
  .views((self) => ({})) // eslint-disable-line @typescript-eslint/no-unused-vars
  .actions((self) => ({})); // eslint-disable-line @typescript-eslint/no-unused-vars

export interface Event extends Instance<typeof EventModel> {}
export interface EventSnapshotOut extends SnapshotOut<typeof EventModel> {}
export interface EventSnapshotIn extends SnapshotIn<typeof EventModel> {}
