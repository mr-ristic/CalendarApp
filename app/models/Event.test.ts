import { getSnapshot, onSnapshot } from 'mobx-state-tree';
import { EventModel } from './Event';
import mockEvents from '../../test/mockEvents';

describe('EventModel test', () => {
  it('should be created with correct props', () => {
    const instance = EventModel.create(mockEvents.data[0]);

    expect(instance).toBeTruthy();
    expect(instance.date).toBe(mockEvents.data[0].date);
  });

  it('should create an instance of a model ', () => {
    const event = mockEvents.data[66];
    const userInstance = EventModel.create(event);

    expect(getSnapshot(userInstance)).toMatchSnapshot();
  });

  it('should update proprety with a action ', () => {
    const event = mockEvents.data[99];
    const userInstance = EventModel.create(event);

    let snapshot: any;

    onSnapshot(userInstance, (newSnapshot) => {
      snapshot = newSnapshot;
    });

    userInstance.setProp('title', 'Meeting with Marko');
    expect(snapshot.title).toBe('Meeting with Marko');
  });
});
