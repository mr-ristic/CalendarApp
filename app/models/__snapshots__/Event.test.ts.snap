// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventModel test should create an instance of a model  1`] = `
{
  "date": "2023-12-04",
  "endTime": "2023-12-04 09:10:57",
  "id": 67,
  "isAllDay": true,
  "location": "Itanha√©m",
  "startTime": "2023-12-04 07:10:57",
  "title": "Barrow County Airport",
  "userId": 17,
}
`;
